api/src/main/java/bisq/api/rest/controller/ApiChatController.java

```diff
@@ -49,17 +50,17 @@ public ApiChatController(ApiApplicationService apiApplicationService) {
     }
 
     @GetMapping(path = "/api/chat/get-public-trade-channels")
-    public List<String> getPublicTradeChannels() {
+    public List<JsonNode> getPublicTradeChannels() {
```

Comment:
What do you think about returning [DTO domain objects](https://www.baeldung.com/entity-to-and-from-dto-for-a-java-spring-application) in controllers. It should give as more flexibility regarding which fields we want to return by API and which to hide. Also swagger documentation would be more robust then.

Reply:
ah, I did answer the argument with the swagger documentation: Thats very true. We probably dont have any other place to put the documentation other than DTOs, when going the swagger way. So the DTO will probably happen soon than later, also when I am looking at the amount of renamings and changes coming in today.. :)

Reply:
Yes, I'm pretty sure we'll finally use DTOs. I don't want to put any pressure on the pace - replacing String with JsonNode is a step in the right direction anyway :)

Reply:
I would like to avoid to get a 3rd data representation. We have already te java objects and the protobuf specs. Once we go live all protobuf data will be hard to change anyway as its used for network and/or persistence, so I think thats good for enforcing some stability for the API as well. Where needed we can tweak with the suggested options but by default creating 100s of DTOs is a lot of overhead and adds lot of maintainance costs.